plans/models.py

```diff
@@ -261,11 +261,20 @@ def get_plan_extended_until(self, plan, pricing):
             return self.expire
         return self.get_plan_extended_from(plan) + timedelta(days=pricing.period)
 
+    def plan_autorenew_at(self):
+        """
+        Helper function which calculates when the plan autorenewal will occur
+        """
+        PLANS_AUTORENEW_BEFORE_DAYS = getattr(settings, 'PLANS_AUTORENEW_BEFORE_DAYS', 0)
+        PLANS_AUTORENEW_BEFORE_HOURS = getattr(settings, 'PLANS_AUTORENEW_BEFORE_HOURS', 0)
+        return self.expire - timedelta(days=PLANS_AUTORENEW_BEFORE_DAYS, hours=PLANS_AUTORENEW_BEFORE_HOURS)
+
     def set_plan_renewal(self, order, has_automatic_renewal=True, **kwargs):
         """
         Creates or updates plan renewal information for this userplan with given order
         """
-        self.recurring.delete()
+        if hasattr(self, 'recurring'):
+            self.recurring.delete()
```

Comment:
I think it would be better to add some logging here. 
Also a single comment would make the operation clear.

For instance why `recurring` object is being deleted and later another one is being created for.

The comment/documentation will make the code much easier to reason about.

Reply:
Also a unit test for this edge case would be a great addition.